### DISCUSSION TITLE
pynev shell boot up messages in Ubuntu

### DISCUSSION THREAD

**Discussion by Trid-collab** [Homebrew/discussions#3110] [Linux] [answered]
have installed pyenv on Ubuntu 20.04 on WSL using Homebrew
I have also configured my shell using the instruction mentioned in the pyenv  site after succesful install of pynev using Homebrew
```
# after any initial comment lines
sed -Ei -e '/^([^#]|$)/ {a \
export PYENV_ROOT="$HOME/.pyenv"
a \
export PATH="$PYENV_ROOT/bin:$PATH"
a \
' -e ':a' -e '$!{n;ba};}' ~/.profile
echo 'eval "$(pyenv init --path)"' >>~/.profile

echo 'eval "$(pyenv init -)"' >> ~/.bashrc
```
but I find that when I rebooting my shell I am getting messages like

```
Command 'pyenv' not found, did you mean:

  command 'pyvenv' from deb python3-venv (3.8.2-0ubuntu2)
  command 'p7env' from deb libnss3-tools (2:3.49.1-1ubuntu1.6)

Try: sudo apt install <deb name>
```
though pyenv seems to working. I am able to install new versions of python, I am getting the above messages while login into my Ubuntu (WSL) 
Is there some mistake I am doing

**Comment by gromgit** [comment, marked as answer]
Check your `~/.profile` and/or `~/.bashrc`. It sounds like you added `eval $(/home/linuxbrew/.linuxbrew/bin/brew shellenv)` _after_ all the `pyenv` stuff. It needs to go _before_ all that, because `pyenv` is in Homebrew's `bin` directory.

**Reply by Trid-collab** [reply]
Thanks it worked. Just one layman's question. I found that `eval $(/home/linuxbrew/.linuxbrew/bin/brew shellenv)` was written in the bottom '.profile` file and the script mentioned in the pyenv site mentioned above was putting everything on the top of .`.profile` file.

I have presently put the following lines at the bottom of my. bashrc and it seems to work. Hope its corect
```
eval $(/home/linuxbrew/.linuxbrew/bin/brew shellenv)

# config for pyenv
# alias brew='env PATH="${PATH//$(pyenv root)\/shims:/}" brew'

export PYENV_ROOT="$HOME/.pyenv"
export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init --path)"

#for pyenv virtualenv
eval "$(pyenv virtualenv-init -)"

#for pipx completions
eval "$(register-python-argcomplete pipx)"
```

**Reply by gromgit** [reply]
I don't use `pyenv`, so I can't comment on that stuff, but a general rule of thumb for ordering lines in `~/.profile` _et al_ is simple: Homebrew's `eval ... shellenv` line _must_ come before running anything that's installed by Homebrew.
